src/main/java/com/iexec/worker/chain/ContributionAuthorizationService.java

```diff
@@ -0,0 +1,67 @@
+package com.iexec.worker.chain;
+
+import com.iexec.common.chain.*;
+import com.iexec.common.contract.generated.IexecHubABILegacy;
+import com.iexec.common.replicate.ReplicateStatusCause;
+import com.iexec.common.security.Signature;
+import com.iexec.common.utils.BytesUtils;
+import com.iexec.common.utils.HashUtils;
+import com.iexec.common.utils.SignatureUtils;
+import com.iexec.worker.config.PublicConfigurationService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Optional;
+
+import static com.iexec.common.replicate.ReplicateStatusCause.*;
+
+
+@Slf4j
+@Service
+public class ContributionAuthorizationService {
+
+    private PublicConfigurationService publicConfigurationService;
+    private HashMap<String, ContributionAuthorization> contributionAuthorizations;
```

Comment:
Map instead of HashMap

Reply:
Rrrrr, automatic IntelliJ isnt smart enought!